apps/gitness/src/pages/repo/repo-commits.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  ListActions,
+  ListPagination,
+  Pagination,
+  PaginationContent,
+  PaginationEllipsis,
+  PaginationItem,
+  PaginationLink,
+  PaginationNext,
+  PaginationPrevious,
+  Spacer,
+  Text
+} from '@harnessio/canary'
+import { BranchSelector, NoData, PaddingListLayout, PullRequestCommits, SkeletonList } from '@harnessio/playground'
+import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
+import { useListBranchesQuery, useListCommitsQuery } from '@harnessio/code-service-client'
+
+const filterOptions = [{ name: 'Filter option 1' }, { name: 'Filter option 2' }, { name: 'Filter option 3' }]
+const sortOptions = [{ name: 'Sort option 1' }, { name: 'Sort option 2' }, { name: 'Sort option 3' }]
+
+export default function RepoCommitsPage() {
+  const repoRef = useGetRepoRef()
+
+  const { data: commitData, isFetching: isFetchingCommits } = useListCommitsQuery({
+    repo_ref: repoRef,
+    queryParams: { page: 0, limit: 10 }
+  })
+  const { data: branches, isFetching: isFetchingBranches } = useListBranchesQuery({
+    repo_ref: repoRef,
+    queryParams: { page: 0, limit: 10 }
+  })
+
+  const renderContent = () => {
+    if (isFetchingCommits) {
+      return <SkeletonList />
+    }
+    // @ts-expect-error remove "@ts-expect-error" once type issue for "content" is resolved
+    if (commitData?.content?.commits?.length) {
```

Comment:
`.content` and the `ts-expect-error` are not needed anymore

Reply:
that PR by @vardanbansal-harness is not merged yet

Reply:
I will remove this as well once my PR gets merged.